### ISSUE TITLE
CREATE DEALS: A not completed deal was posted on the Deals page.

### ISSUE THREAD

**Issue by amateev** [acceptmycrypto/AMC-Web#164]
**Negative Test**: A not completed deal (no confirmation code was sent to a user ) was posted on the Deals page for a user who was authorized later (after creating the second deal and providing a conf code).
In the description area pice of code is showing up.

**Steps to Reproduce**:
condition: the use was not authorized to create deals
1. Created a Deal and go to the step when modal with a contact information form is showing up.
2. Go to Home page, not finishing the module form.
3. Start creating a new deal and finish the process as required with providing confirmation code.
4. Check the Deals page, you will find the first not finished deal 

'Create Deal' is not fully canceled for a not authorized user.
**suggestion**: add a cancel option to the modal that would cancel the Deal.

<img width="1380" alt="screen shot 2019-01-25 at 1 02 17 pm" src="https://user-images.githubusercontent.com/16661480/51772812-dc56a480-20a1-11e9-969b-724475b223bd.png">

**Comment by simonnguyen3054** [comment]
@amateev good catch. I think I need to work on CRUD for listing deal. Allow user to create, edit, and delete deal.

**Comment by simonnguyen3054** [comment]
Im working on this one.

**Comment by simonnguyen3054** [comment]
@amateev I fixed this. Waiting for deployment to heroku

How it works now.
If you created a deal, but not verified, then you're the only one that can see your listing.
On the homepage, your deal card will say pending. 
When you click on it and go to the deal item page, the button will changed to "Verify Your Phone Number"
Clicking the button will show up the modal for you to verify your number
Once verified, all your deals item should be available publicly.
